sklearn/_loss/tests/test_loss.py

```diff
@@ -1048,6 +1048,44 @@ def test_init_gradient_and_hessian_raises(loss, params, err_msg):
         gradient, hessian = loss.init_gradient_and_hessian(n_samples=5, **params)
 
 
+@pytest.mark.parametrize(
+    "loss, params, err_type, err_msg",
+    [
+        (
+            PinballLoss,
+            {"quantile": None},
+            TypeError,
+            "quantile must be an instance of <class 'numbers.Real'>, not <class"
+            " 'NoneType'>.",
+        ),
+        (
+            PinballLoss,
+            {"quantile": 0},
+            ValueError,
+            "quantile == 0, must be > 0.",
+        ),
+        (PinballLoss, {"quantile": 1.1}, ValueError, "quantile == 1.1, must be < 1."),
+        (
+            HalfTweedieLoss,
+            {"power": None},
+            TypeError,
+            "power must be an instance of <class 'numbers.Real'>, not <class"
```

Comment:
If we want nicer messages regarding types, see #22218. 🙂

Reply:
I merged #22218. This test can be adapted then.